### DISCUSSION TITLE
Autoscaling not working as expected

### DISCUSSION THREAD

**Discussion by mwinzie** [eclipse/jkube#1763] [Q&A]
I have hpa.yaml already defined but pods are spining to the maximum amount of replicas. For example my minimum can be one and maximum 2 but 2 pods gets created when it should just be one to begin with.

I included the `jkube.enricher.jkube-controller.replicaCount` and set it to 1 but with maximum replicas of 3, 2 pods get deployed instead of 3. Desired replicas should be 1

**Comment by rohanKanojia** [comment]
Do you have some reproducer project for us to try out? I think Kubernetes APIserver manages autoscaling logic instead of jkube

**Reply by mwinzie** [reply]
Any configuration to set on the pom to set the desired replicas?

**Reply by rohanKanojia** [reply]
I think `jkube.replicas` property should work. Or directly set `<replicas>` field in `<configuration>` > `<resource>` in XML configuration

**Reply by mwinzie** [reply]
will this overwrite the values set in hpa yaml?

**Reply by rohanKanojia** [reply]
No, it overwrites in Deployment yaml

**Reply by mwinzie** [reply]
this works fine. thanks
